core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -471,6 +471,11 @@ oppia.directive('conversationSkin', [
                unwanted scrolling in the new card. */
             $location.hash(null);
             $scope.pendingCardWasSeenBefore = false;
+            if (
+              PlayerTranscriptService.hasEncounteredStateBefore(
+                $scope.displayedCard.getStateName())) {
+              $scope.pendingCardWasSeenBefore = true;
```

Comment:
Instead of:
X = false;
if(Y) {
  X = true;
}

Can we do:
if(Y) {
  X = false;
}

Not sure whether it's possible or not! Can you please check? (if required we can add `else` block too!)

Reply:
Do you mean we don't need `X = true` in the latter case ?

Reply:
@bansalnitish, I was asking you to follow the pattern

if(something) {
  do this;
} else {
  do that;
}

I fell this is more readable. I'm approving this PR!